File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/Sample Code/PrepData.py
## Function to prepare the automotive data
def prep_auto(df, col_names):
    import pandas as pd
    import numpy as np
    
    ## Assign names to columns
    df.columns = col_names
    
    ## Drop unneeded columns
    drop_list = ['symboling', 'normalizedlosses', 'makeid']    
    df.drop(drop_list, axis = 1, inplace = True)
    
    ## Remove rows with missing values    
    df = df[~pd.isnull(df).any(axis=1)]
    
    ## Add a log transformed column for price
    df['lnprice'] = np.log(df['price'].as_matrix()) 
    
    ## Remove duplicate rows
    df.drop_duplicates(inplace = True) 
    
    ## Create a column with new levels for the number of cylinders
    df['cylinders'] = ['four-or-less' if x in ['two', 'three', 'four'] else 
                                 ('five-six' if x in ['five', 'six'] else 
                                 'eight-twelve') for x in df['cylinders']]    
    return df     

def azureml_main(df):   
    ## Define column names
    col_names = ['symboling', 'normalizedlosses', 'makeid', 'fueltype', 'aspiration', 'doors',
       'body', 'drive', 'engineloc', 'wheelbase',
       'length', 'width', 'height', 'weight', 'enginetype',
       'cylinders', 'enginesize', 'fuelsystem', 'bore', 'stroke',
       'compression', 'horsepower', 'rpm', 'citympg',
       'highwaympg', 'price', 'make']
       
    ## Call function to prep auto data and return       
    return prep_auto(df, col_names)
      

File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/Sample Code/PrepData.R
## Function to clean and prepare the auto data
prep.auto <- function(df, col.names){
  require(dplyr) ## Make sure dplyr is loaded
  
  ## set the column names. 
  names(df) <- col.names
  
  ## Eliminate unneeded columns
  df <- df[,!(names(df) %in% 
                c('symboling', 'normalizedlosses', 'makeid'))]
 
  ## Coerce some character columns to numeric 
  ## Uncomment if NOT in Azure ML
  # cols <- c('price', 'bore', 'stroke', 'horsepower', 'rpm')
  # df[, cols] <- lapply(df[, cols], as.numeric)

  ## Add a log transformed column for price using dplyr mutate
  df <- df %>% mutate(lnprice = log(price))
  
  ## Remove rows with NAs 
  df <- df[complete.cases(df), ]
  
  ## Remove duplicate rows
  df <- df %>% filter (! duplicated(df,))
  
  ## Consolidate the number of cylinders
  df <- df %>%
    mutate(cylinders = ifelse(cylinders %in% c("four", "three", "two"), "three-four",
           ifelse(cylinders %in% c("five", "six"), "five-six", "eight-twelve")))
  df
}

## Define column names
col.names <- c('symboling', 'normalizedlosses', 'makeid', 'fueltype', 'aspiration', 'doors',
               'body', 'drive', 'engineloc', 'wheelbase',
               'length', 'width', 'height', 'weight', 'enginetype',
               'cylinders', 'enginesize', 'fuelsystem', 'bore', 'stroke',
               'compression', 'horsepower', 'rpm', 'citympg',
               'highwaympg', 'price', 'make')

## R code to prep the auto data
autos.price <- maml.mapInputPort(1) # read autos data frame from port 1
out <- prep.auto(autos.price, col.names) # Transform the data
maml.mapOutputPort("out") ## Output the prepared data frame